sdk/src/binary.rs

```diff
@@ -0,0 +1,283 @@
+        } else {
+            Err(())
+        }
+    }
+}
+
+impl From<Binary> for RawVal {
+    #[inline(always)]
+    fn from(v: Binary) -> Self {
+        v.0.into()
+    }
+}
+
+impl From<Binary> for EnvVal {
+    #[inline(always)]
+    fn from(v: Binary) -> Self {
+        v.0.into()
+    }
+}
+
+impl From<Binary> for EnvObj {
+    #[inline(always)]
+    fn from(v: Binary) -> Self {
+        v.0
+    }
+}
+
+impl Binary {
+    #[inline(always)]
+    unsafe fn unchecked_new(obj: EnvObj) -> Self {
+        Self(obj)
+    }
+
+    #[inline(always)]
+    fn env(&self) -> &Env {
+        self.0.env()
+    }
+
+    #[inline(always)]
+    pub fn new(env: &Env) -> Binary {
+        let obj = env.binary_new().in_env(env);
+        unsafe { Self::unchecked_new(obj) }
+    }
+
+    #[inline(always)]
+    pub fn put(&mut self, i: u32, v: u8) {
+        let v32: u32 = v.into();
+        self.0 = self
+            .env()
+            .binary_put(self.0.to_tagged(), i.into(), v32.into())
+            .in_env(self.env());
+    }
+
+    #[inline(always)]
+    pub fn get(&self, i: u32) -> u8 {
+        let res32: u32 = self
+            .env()
+            .binary_get(self.0.to_tagged(), i.into())
+            .try_into()
+            .unwrap();
+        res32.try_into().unwrap()
+    }
+
+    #[inline(always)]
+    pub fn del(&mut self, i: u32) {
+        self.0 = self
+            .env()
+            .binary_del(self.0.to_tagged(), i.into())
+            .in_env(self.env());
+    }
+
+    #[inline(always)]
+    pub fn len(&self) -> u32 {
+        self.env()
+            .binary_len(self.0.to_tagged())
+            .try_into()
+            .unwrap()
+    }
+
+    pub fn push(&mut self, x: u8) {
+        let x32: u32 = x.into();
+        self.0 = self
+            .env()
+            .binary_push(self.0.to_tagged(), x32.into())
+            .in_env(self.env());
+    }
+
+    pub fn pop(&mut self) {
+        self.0 = self.env().binary_pop(self.0.to_tagged()).in_env(self.env());
+    }
+
+    pub fn front(&self) -> u8 {
+        let res32: u32 = self
+            .env()
+            .binary_front(self.0.to_tagged())
+            .try_into()
+            .unwrap();
+        res32.try_into().unwrap()
+    }
+
+    pub fn back(&self) -> u8 {
+        let res32: u32 = self
+            .env()
+            .binary_back(self.0.to_tagged())
+            .try_into()
+            .unwrap();
+        res32.try_into().unwrap()
+    }
+
+    pub fn insert(&mut self, i: u32, x: u8) {
+        let x32: u32 = x.into();
+        self.0 = self
+            .env()
+            .binary_insert(self.0.to_tagged(), i.into(), x32.into())
+            .in_env(self.env());
+    }
+
+    pub fn append(&mut self, other: &Binary) {
+        self.0 = self
+            .env()
+            .binary_append(self.0.to_tagged(), other.0.to_tagged())
+            .in_env(self.env());
+    }
+}
+
+#[derive(Clone)]
+#[repr(transparent)]
+pub struct FixedLengthBinary<const N: u32>(Binary);
```

Comment:
We should impl the `put` fn and pass through to the underlying `Binary`.

Reply:
Done, https://github.com/stellar/rs-stellar-contract-sdk/pull/148#discussion_r907531873.